pkg/apis/networking/ports.go

```diff
@@ -40,6 +40,11 @@ const (
 	ServicePortNameHTTPS = "https"
 )
 
+var (
+	// AppProtocolH2C is the name of the external port of the service for HTTP/2
+	AppProtocolH2C = "kubernetes.io/h2c"
```

Comment:
How about `kubernetes.io/ws` ?

Reply:
Could we somehow make clear that this is gateway api related?

Reply:
> How about kubernetes.io/ws? Is there a plan for this?

Ah, good point. I think I may have misinterpretted Dave's sentence on WS here: https://github.com/knative/serving/issues/14569


He says:
> For websocket - this in theory should just be on by default (contour toggled it on for Gateway API). There might be other implementations that have this off but let's revisit that when we find one that doesn't

Which when I first read it, I thought, leave it and come back to it, but perhaps what he's really saying is put it on always?

Reply:
> Could we somehow make clear that this is gateway api related?

Sure, like perhaps calling it `GatewayAPIAppProtocolH2C` or something?

Reply:
ah, you know, reading [Dave's GEP](https://github.com/kubernetes-sigs/gateway-api/blob/main/geps/gep-1911/index.md#default-protocols), maybe we don't need to do anything for WS, since upgrades happen by default?

@dprotaso what were you imagining here?

also, how does one indicate that a Knative Service should be a websocket?

Reply:
> Sure, like perhaps calling it GatewayAPIAppProtocolH2C or something?

These constants are technically Kubernetes constants - I don't think we should prefix them with the Gateway API

We can link the KEP so it's clear

https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/3726-standard-application-protocols#new-standard-protocols

Honestly, websocket is a weird one cause Contour should have probably not disabled it. For now we can skip it since I'd imagine most implementations support the upgrade flow via HTTP

So we really need to set the h2c protocol when the user specifies it in their port name (the way it's used for grpc services)

eg.
https://github.com/knative/docs/blob/dab74765e0f0c60789fe97208aa0b0a1e23eb9cd/code-samples/serving/grpc-ping-go/service.yaml#L12C13-L12C22

Reply:
If we need websocket support we would have to upgrade the runtime contract to define a constants users would use

https://github.com/knative/specs/blob/main/specs/serving/runtime-contract.md#protocols-and-ports

Reply:
so, to summarize:
* keep the variable name, but link the KEP in the comment
* don't move forward with WS right now

is that correct?

As far as this comment:
> So we really need to set the h2c protocol when the user specifies it in their port name (the way it's used for grpc services)

My other PR (https://github.com/knative/serving/pull/14757) is where that would happen, right?

Reply:
> My other PR (knative/serving#14757) is where that would happen, right?

Yeah